### PR TITLE
Add devnet

### PR DESCRIPTION
**Pull request by JorikSchellekens**
The new Dockerfile is compatible with the [old single container setup](https://github.com/NethermindEth/warp/tree/devnet-docs#docker-whale) so the instructions doesn't need to change

### CODE REVIEW

**Review by swapnilraj — APPROVED** [review]

File: USING-DEVNETS.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export STARKNET_GATEWAY_URL=http://localhost:5050
+export STARKNET_FEEDER_GATEWAY_URL=htp://localhost:5050
+export STARKNET_ACCOUNT_DIR=~/.starknet_accounts_devnet
+```
+
+It's a good idea to then launch the devnet with the following
+
+```
+starknet-devent --timeout 10000 --seed 0
```
Comment (piwonskp) [inline USING-DEVNETS.md:15]: Typo

File: USING-DEVNETS.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Here are some useful env variables to have set if you are targeting a local StarkNet devnet
+
+```
+export STARKNET_NETWORK=alpha-goerli
+export STARKNET_WALLET=starkware.starknet.wallets.open_zeppelin.OpenZeppelinAccount
+
+export STARKNET_GATEWAY_URL=http://localhost:5050
```
Comment (swapnilraj) [inline USING-DEVNETS.md:7]: Its just GATEWAY_URL and FEEDER_GATEWAY_URL no prefix neede

**Comment by swapnilraj** [comment]
Took the liberty to address the comments

File: USING-DEVNETS.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+To use starknet devnet you have to export evironment variables first:
+
+```bash
+export $(cat development.env | xargs)
```
Comment (swapnilraj) [inline USING-DEVNETS.md:4]: `source development.env`

File: development.env [hunk 4]
```diff
@@ -0,0 +1,7 @@
+STARKNET_NETWORK=alpha-goerli
+STARKNET_WALLET=starkware.starknet.wallets.open_zeppelin.OpenZeppelinAccount
+STARKNET_ACCOUNT_DIR=~/.starknet_accounts_devnet
```
Comment (swapnilraj) [inline development.env:3]: If we set `STARKNET_ACCOUNT_DIR=$PWD` then we will not need to copy the account json file and this way we also avoid copying over the original account json

File: USING-DEVNETS.md (hunk 3 above)
Comment (piwonskp) [reply USING-DEVNETS.md]: Source creates shell variables instead of env variables as source expects the file to be a shell script

Comment (piwonskp) [reply USING-DEVNETS.md]: Ok, there is however an option to do it with source and still preserve the .env syntax

File: readme.md [hunk 5]
```diff
@@ -5,6 +5,37 @@
...
+
+Alternatively for warp developers:
+
+```bash
+docker-compose exec warp npx ts-node src transpile example_contracts/ERC20.sol
+```
+
+It's best to copy the contract/repo to the warp directory so it is available in container via volume. Use contract's paths relative to warp root. For example, assuming you've copied your project to `warp/projects/myproject` you can replace `example_contracts/ERC20.sol` with `projects/myproject/mycontract.sol` in the above commands.
+
+### Deploy to devnet
```
Comment (swapnilraj) [inline readme.md:32]: We should move this to the dev section, there is a command for compiling with warp
`warp compile` we should use that.

In a follow up PR can you also make it possible to deploy to the devnet using warp deploy and then replace starknet deploy with warp deploy

Comment (piwonskp) [reply readme.md:32]: Not sure what you mean by dev section. If [that](https://github.com/NethermindEth/warp#warp-installation-method-2-from-sourcefor-devs) then the section is not describing docker-compose usage

**Pull request merged**